lua/completion.lua

```diff
@@ -98,7 +98,7 @@ local function applyAddtionalTextEdits(completed_item)
   if completed_item.user_data.lsp ~= nil then
     local item = completed_item.user_data.lsp.completion_item
     -- vim-vsnip have better additional text edits...
-    if vim.fn.exists('g:loaded_vsnip_integ') == 1 then
```

Comment:
I think that the `== 1` is mandatory here, as VimL functions return an integer.

Reply:
Result may be an integer, but according to documentation, **any** value other than zero is considered True, and only 0 is considered False:

```
                                                       exists()                                            
exists({expr})  The result is a Number, which is TRUE if {expr} is                                          
                defined, zero otherwise.
```

So which integer doesn't really matters, as long as it can be interpreted the same way boolean expressions are interpreted in `C` language.

Restricting the comparison to `== 1` may lead to unexpected behaviors if for instance, on runtime, the `exists()` function decides to return `2`; which according to documentation is also a valid return value when the expression **is** defined.

Reply:
But `0` is `true` in Lua.

Reply:
In any case, comparing to a value, the line should be:

```vim
if vim.fn.exists('g:loaded_vsnip_integ') != 0 then
```

But again, using function's result as a boolean value is correct according to the documentation. No further comparison to integer values is really necessary.

Reply:
> But `0` is `true` in Lua. You could do `if XX ~= 0`

Ok, true, corrected the PR